### ISSUE TITLE
Rootless 'podman create' fails to marshal JSON in v1.0 branch

### ISSUE THREAD

**Issue by debarshiray** [containers/libpod#2139]
/kind bug

**Description**

Rootless `podman create` is broken in `v1.0` (commit fbed1650fbcc32fb), but works in `master` (commit 9368c24be6f289d2):
```
$ sudo rm -rf ~/.local/share/containers
$ rm ~/.config/containers/*.conf
$ $ podman create fedora:29 /bin/sh
Trying to pull docker.io/fedora:29...Getting image source signatures
Copying blob e1736333d405: 85.71 MiB / 85.71 MiB [==========================] 5s
Copying config 8bd04269a51b: 2.20 KiB / 2.20 KiB [==========================] 0s
Writing manifest to image destination
Storing signatures
error marshalling container d46588ee567bf043479137b08c02b39587f85b4b520fabf51ba0bbcf0c499f08 config to JSON: json: error calling MarshalJSON for type *libpod.ContainerConfig: unexpected end of JSON input
```

**Output of `podman version`:**

```
Version:       0.12.2-dev
Go Version:    go1.11.2
Git Commit:    "fbed1650fbcc32fb514293d18daa26b5e8029876"
Built:         Fri Jan 11 15:12:16 2019
OS/Arch:       linux/amd64
```

**Output of `podman info`:**

```
host:
  BuildahVersion: 1.6-dev
  Conmon:
    package: podman-0.12.1.2-1.git9551f6b.fc29.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 1.12.0-dev, commit: 67ab7549b44484cc3f201d7bb2b58b922f8edc24'
  Distribution:
    distribution: fedora
    version: "29"
  MemFree: 1826705408
  MemTotal: 8127053824
  OCIRuntime:
    package: runc-1.0.0-66.dev.gitbbb17ef.fc29.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc6+dev
      commit: ead425507b6ba28278ef71ad06582df97f2d5b5f
      spec: 1.0.1-dev
  SwapFree: 4133482496
  SwapTotal: 4133482496
  arch: amd64
  cpus: 4
  hostname: kolache
  kernel: 4.19.12-301.fc29.x86_64
  os: linux
  rootless: true
  uptime: 4h 16m 32.7s (Approximately 0.17 days)
insecure registries:
  registries: []
registries:
  registries:
  - docker.io
  - registry.fedoraproject.org
  - quay.io
  - registry.access.redhat.com
  - registry.centos.org
store:
  ConfigFile: /home/rishi/.config/containers/storage.conf
  ContainerStore:
    number: 0
  GraphDriverName: overlay
  GraphOptions:
  - overlay.mount_program=/usr/bin/fuse-overlayfs
  GraphRoot: /home/rishi/.local/share/containers/storage
  GraphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
  ImageStore:
    number: 1
  RunRoot: /run/user/1000
```

**Comment by mheon** [comment]
Not rootless specific

**Comment by mheon** [comment]
Fixed by #2140

**Comment by debarshiray** [comment]
Yes, #2140 fixes this for me. Thanks @mheon 

I wonder why `test/e2e/create_test.go` didn't flag this earlier.

**Comment by mheon** [comment]
We're still getting the tests up and running on the 1.0 branch, which is probably why. Need to do a full test run before we cut the release and package up.

**Comment by mheon** [comment]
Fixed via #2140
